stdlib/Test/test/runtests.jl

```diff
@@ -797,29 +793,30 @@ end
     @test startswith(fails[4].value, "ErrorException")
 end
 
-function newfunc()
-    42
-end
-@deprecate oldfunc newfunc
+let code = quote
+        function newfunc()
+            42
+        end
+        @deprecate oldfunc newfunc
 
-@testset "@test_deprecated" begin
-    @test_deprecated oldfunc()
+        @testset "@test_deprecated" begin
+            @test_deprecated oldfunc()
+            @test Base.JLOptions().depwarn == 1
 
-    # Expression passthrough
-    if Base.JLOptions().depwarn != 2
-        @test (@test_deprecated oldfunc()) == 42
+            @test (@test_deprecated oldfunc()) == 42
 
-        fails = @testset NoThrowTestSet "check that @test_deprecated detects bad input" begin
-            @test_deprecated newfunc()
-            @test_deprecated r"Not found in message" oldfunc()
+            fails = @testset NoThrowTestSet "check that @test_deprecated detects bad input" begin
+                @test_deprecated newfunc()
+                @test_deprecated r"Not found in message" oldfunc()
+            end
+            @test length(fails) == 2
+            @test fails[1] isa Test.LogTestFailure
+            @test fails[2] isa Test.LogTestFailure
         end
-        @test length(fails) == 2
-        @test fails[1] isa Test.LogTestFailure
-        @test fails[2] isa Test.LogTestFailure
-    else
-        @warn """Omitting `@test_deprecated` tests which can't yet
-                 be tested in --depwarn=error mode"""
     end
+    incl = "include($(repr(joinpath(@__DIR__, "nothrow_testset.jl"))))"
+    cmd = `$(Base.julia_cmd()) --startup-file=no --depwarn=yes -e 'using Test' -e $incl -e $code`
```

Comment:
The fact that `-e $code` works in backticks and is completely safe... :heart_eyes: 

(Yes, I've been bash scripting today)

Reply:
I think it can often still be a bit more reliable to put the content into stdin (via pipeline below). But the maximum here is `getconf ARG_MAX`, or 2MB on linux, 256KB on most posix, and 32k on Windows, and we're very well inside those limits here.